### PR TITLE
fix(admin): load app custom with jsx and allow more extensions

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Upgrade the laodFile util to support required feature.

### Why is it needed?

Fix an issue with custom app code

### How to test it?

create a src/admin/app.(ts|tsx|js|jsx) file with jsx code in it

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

**Comment by alexandrebodin** [comment]
@joshuaellis  I'm wondering why we even bother using esbuild to load that file ? we just need it's path to generate the `.strapi/app.js` file I think

**Comment by joshuaellis** [comment]
> @joshuaellis I'm wondering why we even bother using esbuild to load that file ? we just need it's path to generate the `.strapi/app.js` file I think

I don't think we need to in retrospective, although in the future we might want to run validations against it but we can do that another time. I think using `pathExists` from the `files.ts` will suffice to assert it's the file we expect :)

File: packages/core/admin/_internal/node/core/files.ts [hunk 1]
```diff
@@ -18,7 +18,11 @@ const pathExists = async (path: string) => {
  */
 const loadFile = async (path: string): Promise<undefined | any> => {
   if (await pathExists(path)) {
-    const esbuildOptions = { extensions: ['.js', '.mjs', '.ts'] };
+    const esbuildOptions: Parameters<typeof register>[0] = {
+      extensions: ['.js', '.mjs', '.ts', '.jsx', '.tsx'],
+      jsx: 'automatic',
+      loader: 'jsx',
+    };
```
Comment (joshuaellis) [inline packages/core/admin/_internal/node/core/files.ts:25]: Do we still need this change? 👀

Comment (alexandrebodin) [reply packages/core/admin/_internal/node/core/files.ts]: Removed it that should be enough to laod webpack & plugins conf only

**Review by joshuaellis — APPROVED** [review]

File: packages/core/admin/_internal/node/core/admin-customisations.ts [hunk 2]
```diff
@@ -12,16 +12,14 @@ interface AdminCustomisations {
...
 }
 
 const loadUserAppFile = async (appDir: string): Promise<AppFile | undefined> => {
   for (const file of ADMIN_APP_FILES) {
     const filePath = path.join(appDir, 'src', 'admin', file);
-    const configFile = await loadFile(filePath);
 
-    if (configFile) {
-      return { path: filePath, config: configFile };
+    if (fs.existsSync(filePath)) {
```
Comment (yuan1238y) [inline packages/core/admin/_internal/node/core/admin-customisations.ts:21]: Why use `existsSync` in `async` function ?

**Pull request merged**